src/main/java/com/aphyr/riemann/client/Promise.java

```diff
@@ -75,6 +99,11 @@ public Object unsafeDeref(final long time,
     }
   }
 
+  @Override
+  public void setCallback(Callback<T> c) {
```

Comment:
Hmm right. So should it just check if the value's already been delivered and just call the callback immediately if so? Would have to do some synchronizing to prevent the possibility of calling twice there...